src/shogun/machine/gp/SingleFITCLaplacianBase.h

```diff
@@ -0,0 +1,269 @@
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * The views and conclusions contained in the software and documentation are those
+ * of the authors and should not be interpreted as representing official policies,
+ * either expressed or implied, of the Shogun Development Team.
+ *
+ */
+
+#ifndef CSINGLEFITCLAPLACIANBASE_H
+#define CSINGLEFITCLAPLACIANBASE_H
+
+#include <shogun/lib/config.h>
+
+#ifdef HAVE_EIGEN3
+
+#include <shogun/machine/gp/FITCInferenceBase.h>
+#include <shogun/lib/Lock.h>
+
+namespace shogun
+{
+
+/** @brief The Fully Independent Conditional Training inference base class
+ * for Laplace and regression for 1-D labels (1D regression and binary classification)
+ *
+ * This base class implements the (explicit) derivatives of negative log marginal likelihood
+ * wrt hyperparameter for FITC regression and FITC single Laplace.
+ * For FITC single Laplace, we can compute further implicit derivatives.
+ * For FITC regression, these explicit derivatives are the full derivatives. 
+ *
+ * For more details, see Quiñonero-Candela, Joaquin, and Carl Edward Rasmussen.
+ * "A unifying view of sparse approximate Gaussian process regression."
+ * The Journal of Machine Learning Research 6 (2005): 1939-1959.
+ *
+ * This specific implementation was inspired by the infFITC.m and infFITC_Laplace.m file
+ * in the GPML toolbox.
+ *
+ */
+class CSingleFITCLaplacianBase: public CFITCInferenceBase
+{
+public:
+	/** default constructor */
+	CSingleFITCLaplacianBase();
+
+	/** constructor
+	 *
+	 * @param kernel covariance function
+	 * @param features features to use in inference
+	 * @param mean mean function
+	 * @param labels labels of the features
+	 * @param model likelihood model to use
+	 * @param inducing_features features to use
+	 */
+	CSingleFITCLaplacianBase(CKernel* kernel, CFeatures* features,
+			CMeanFunction* mean, CLabels* labels, CLikelihoodModel* model,
+			CFeatures* inducing_features);
+
+	virtual ~CSingleFITCLaplacianBase();
+
+	/** returns the name of the inference method
+	 *
+	 * @return name SingleFITCLaplacianBase 
+	 */
+	virtual const char* get_name() const { return "SingleFITCLaplacianBase"; }
+
+	/** set kernel
+	 *
+	 * @param kern kernel to set
+	 */
+	virtual void set_kernel(CKernel* kern);
+protected:
+	/** check whether the provided kernel can
+	 * compute the gradient wrt inducing features
+	 */
+	virtual void check_fully_FITC();
+
+	/** compute matrices which are required to compute negative log marginal
+	 * likelihood full derivatives wrt  cov-like hyperparameter
+	 *
+	 * Note that
+	 * scale, which is a hyperparameter in inference_method, is a cov-like hyperparameter
+	 * hyperparameters in cov function are cov-like hyperparameters
+	 *
+	 * @param the gradient wrt hyperparameter related to cov-like hyperparameter
```

Comment:
Could you document all parameters here?
doxygen will complain otherwise?
Maybe reference the math somewhere?